source/common/http/http2/codec_impl.h

```diff
@@ -176,6 +176,17 @@ class ConnectionImpl : public virtual Connection, protected Logger::Loggable<Log
     const Network::Address::InstanceConstSharedPtr& connectionLocalAddress() override {
       return parent_.connection_.localAddress();
     }
+    absl::string_view responseDetails() override { return details_; }
+
+    void setDetails(absl::string_view details) {
+      // It is probably a mistake to call setDetails() twice, so
+      // assert that details_ is empty.
+      ASSERT(details_.empty());
+
+      // make a copy of details to decouple the lifetime of the two
+      // objects details and details_.
+      details_ = static_cast<std::string>(details);
```

Comment:
For perf reasons it seems like you could just stipulate/assume that the passed strings are constant, but I don't feel strongly about it. Up to you.

Reply:
Yeah, my original code agreed with you, but yanavlasov requested this change, and now both you and Alyssa seem to agree with my initial code.  I'll talk to Alyssa and decide what to do.

FWIW, the http1 codec assumes that the details_ string is static, so if we decide to be paranoid, I should fix the http1 codec as well.

Reply:
> I'll talk to Alyssa and decide what to do.

Alyssa is out this week. My preference us to assume they are static strings, but I agree that if you and @yanavlasov decide otherwise we should be consistent across both codecs here.

Reply:
I have pushed a patch tentatively rolling back to static strings, while I sort the issue out with @yanavlasov